app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -189,8 +195,22 @@ class ConversationViewModel @Inject constructor(
         onDialogDismissed()
     }
 
-    private fun List<com.wire.kalium.logic.data.message.Message>.toUIMessages(): List<Message> {
+    private fun List<MemberDetails>.findSender(senderId: UserId): MemberDetails? = firstOrNull { member ->
+        when (member) {
+            is MemberDetails.Other -> member.otherUser.id == senderId
+            is MemberDetails.Self -> member.selfUser.id == senderId
+        }
+    }
+
+    private val MemberDetails.name
+        get() = when (this) {
+            is MemberDetails.Other -> this.otherUser.name
+            is MemberDetails.Self -> this.selfUser.name
+        }
+
+    private fun List<com.wire.kalium.logic.data.message.Message>.toUIMessages(members: List<MemberDetails>): List<Message> {
```

Comment:
Although this is only being used here, I would suggest having it in a separate file for its extensions, more preferably on the data class file, that way if it is ever needed somewhere else in the code we know it exists.

Reply:
As the class is defined in Kalium, I've extracted the extension function to another file.